src/user/mod.rs

```diff
@@ -59,30 +60,21 @@ impl UserMetadata {
         }
     }
 
-    pub fn get_user_result(&mut self, a: &UserAction) -> String {
-        query_user_input(
+    pub(crate) fn get_user_prompt(&mut self, a: &UserAction) -> anyhow::Result<UserPrompt> {
+        let input = query_user_input(
             a.to_string().lines().map(str::to_string).collect(),
             self.get_prompt_items(a)
                 .into_iter()
                 .map(|p| p.to_string())
                 .collect(),
             &UserAction::ModifyExisting == a,
-        )
-    }
-}
-
-impl Default for UserMetadata {
-    fn default() -> Self {
-        UserMetadata {
-            root: None,
-            inputs: None,
-            filename: None,
-        }
+        )?;
+        Ok(UserPrompt::from_str(&input).unwrap())
```

Comment:
`.unwrap()` is inappropriate, try to transform the error instead, we don't want to panic on invalid/misspelled commands.

Reply:
I should've made a comment; `UserPrompt`'s `FromStr` impl cannot fail since the last case `Other(String)` will catch everything.

Reply:
I suppose the `.unwrap` could be replaced with an infallible match, which would fix that.

Reply:
Hmm, it looks like `parse_display` sets `ParseError(&'static str)` as the `Err` type of `FromStr`, so just for correctness we can go with catching any errors (although there shouldn't be one).

Reply:
ok, I tried that, won't work.

Reply:
An `.expect("...")` with a good error message is usually enough.